src/Microsoft.PowerShell.ConsoleHost/host/msh/ConsoleHostUserInterface.cs

```diff
@@ -645,13 +645,19 @@ private void WriteToConsole(ConsoleColor foregroundColor, ConsoleColor backgroun
 
                 try
                 {
-                    WriteToConsole(text, transcribeResult: true, newLine);
+                    WriteToConsole(text, transcribeResult: true, newLine: false);
                 }
                 finally
                 {
                     RawUI.ForegroundColor = fg;
                     RawUI.BackgroundColor = bg;
                 }
+
+                if (newLine)
+                {
+                    // write the newline after resetting the colors so there isn't a buffer wide colored line when buffer scrolls
+                    WriteLineToConsole(string.Empty, transcribeResult: true);
+                }
```

Comment:
I wonder why it is not internally in WriteLineToConsole.

Reply:
This was a mistake, `WriteLineToConsole()` actually calls this method